app/org/sagebionetworks/bridge/models/schedules/TaskScheduler.java

```diff
@@ -0,0 +1,26 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import java.util.List;
+import java.util.Map;
+
+import org.joda.time.DateTime;
+
+public interface TaskScheduler {
+
+    /**
+     * Get all the tasks for the schedule that start before up to the "until" date and time passed to the method. Tasks
```

Comment:
I will, but as it isn't done yet, it's more like writing a design document in the javadocs. I've done that as well as I'm able in the wiki, and will need to return to the javadocs when the tasks portion of this work is done.